frontend/app/route-helpers/renew-ita-route-helpers.server.ts

```diff
@@ -0,0 +1,127 @@
+import type { Session } from '@remix-run/node';
+import { redirect } from '@remix-run/node';
+import type { Params } from '@remix-run/react';
+import { isRedirectResponse, isResponse } from '@remix-run/server-runtime/dist/responses';
+
+import { loadRenewState, saveRenewState } from './renew-route-helpers.server';
+import type { RenewState } from './renew-route-helpers.server';
+import { getLogger } from '~/utils/logging.server';
+import { getPathById } from '~/utils/route-utils';
+
+interface LoadRenewItaStateArgs {
+  params: Params;
+  request: Request;
+  session: Session;
+}
+
+/**
+ * Loads renew ITA state.
+ * @param args - The arguments.
+ * @returns The loaded ITA state.
+ */
+export function loadRenewItaState({ params, request, session }: LoadRenewItaStateArgs) {
+  const log = getLogger('renew-ita-route-helpers.server/loadRenewItaState');
+  const { pathname } = new URL(request.url);
+  const renewState = loadRenewState({ params, session });
+
+  if (renewState.typeOfRenewal !== 'adult-child') {
+    throw redirect(getPathById('public/renew/$id/type-application', params));
+  }
+
+  // Redirect to the confirmation page if the application has been submitted and
+  // the current route is not the confirmation page.
+  const confirmationRouteUrl = getPathById('public/renew/$id/adult-child/confirmation', params);
```

Comment:
I think all these paths with `/adult-child` in them should be changed (to something more specific to ITA?) because `/adult-child` would be reserved for "(B) Adult or family" flow in the Figma.